File: 300-longest-increasing-subsequence/longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n=nums.size();
        int currl[n];
        fill_n(currl,n,1);
        int ans=1;
        for(int i=0;i<nums.size();i++){
            for(int j=0;j<i;j++){
                if(nums[j]<nums[i]){
                    if(currl[j]+1>currl[i]){
                        currl[i]=currl[j]+1;
                    }
                }
             ans=max(ans,currl[i]);   
            }
        }return ans;
    }
};

File: 300-longest-increasing-subsequence/longest-increasing-subsequence.py
class Solution(object):
    def lengthOfLIS(self, arr):
        n = len(arr)
        dp = [1] * n

        for i in range(1, n):
            for j in range(0, i):
                if arr[i] > arr[j]:
                    dp[i] = max(dp[i], dp[j] + 1)

        return max(dp)

        